creator-node/src/serviceRegistry.js

```diff
@@ -44,64 +52,197 @@ class ServiceRegistry {
     await this.blacklistManager.init()
 
     if (!config.get('isUserMetadataNode')) {
-      this.libs = await initAudiusLibs()
-
-      this.snapbackSM = new SnapbackSM(this.libs)
-      await this.snapbackSM.init()
+      this.libs = await this._initAudiusLibs()
     }
 
     this.monitoringQueue.start()
   }
-}
 
-/** Private helper used by ServiceRegistry for initializing libs */
-const initAudiusLibs = async () => {
-  const ethWeb3 = await AudiusLibs.Utils.configureWeb3(
-    config.get('ethProviderUrl'),
-    config.get('ethNetworkId'),
-    /* requiresAccount */ false
-  )
-  const dataWeb3 = await AudiusLibs.Utils.configureWeb3(
-    config.get('dataProviderUrl'),
-    /* chainNetworkId */ null,
-    /* requiresAccount */ false
-  )
-
-  if (!ethWeb3 || !dataWeb3) {
-    throw new Error('Failed to init audiusLibs due to web3 configuration error')
+  /**
+   * Some services require the node server to be running in order to initialize. Run those here.
+   * Specifically:
+   *  - recover node L1 identity (requires node health check from server to return success)
+   *  - initialize SnapbackSM service (requires node L1 identity)
+   *  - register node on L2 URSM contract (requires node L1 identity)
+   */
+  async initServicesThatRequireServer () {
+    if (config.get('isUserMetadataNode')) {
+      return
+    }
+
+    // Cannot progress without recovering spID from node's record on L1 ServiceProviderFactory contract
+    await this._recoverNodeL1Identity()
```

Comment:
what happens if one of these fails?

Reply:
i see, each is internally retried indefinitely

Reply:
yee

Reply:
added a comment tho